source/MaterialXGenOgsXml/OgsXmlGenerator.cpp

```diff
@@ -142,26 +128,21 @@ namespace
         }
     }
 
-    void xmlSetProperty(pugi::xml_node& prop, const string& name, const string& variable, const string& defaultFlags = EMPTY_STRING)
+    void xmlSetProperty(pugi::xml_node& prop, const string& name, const string& variable, const string& parameterFlags = EMPTY_STRING)
     {
         prop.append_attribute(NAME) = variable.c_str();
         const auto semantic = OGS_SEMANTICS_MAP.find(name);
         if (semantic != OGS_SEMANTICS_MAP.end())
         {
             prop.append_attribute(SEMANTIC) = semantic->second;
         }
-        const auto flag = OGS_FLAGS_MAP.find(name);
-        if (flag != OGS_FLAGS_MAP.end())
```

Comment:
Don't consider exceptions, just use the flags common to the whole type of inputs, in our case vertex inputs.